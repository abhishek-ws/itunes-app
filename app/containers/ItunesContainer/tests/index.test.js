/**
 *
 * Tests for ItunesContainer
 *
 */
import React from 'react';
import { timeout, renderProvider } from '@utils/testUtils';
import { fireEvent } from '@testing-library/dom';
import { ItunesContainerTest as ItunesContainer } from '../index';
import ReturnWithRouter from '@utils/returnRouter';

function ItunesWrapper(props) {
  return (
    <ReturnWithRouter>
      <ItunesContainer {...props} />
    </ReturnWithRouter>
  );
}

describe('ItunesContainer Tests', () => {
  let mockDispatchSearch;

  beforeEach(() => {
    mockDispatchSearch = jest.fn();
  });

  it('should render and match to the snapshot', () => {
    const { baseElement } = renderProvider(<ItunesWrapper dispatchSearchSongs={mockDispatchSearch} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dispatchClearGridData on empty change in SerchBox', async () => {
    const searchSongsSpy = jest.fn();
    const clearGridDataSpy = jest.fn();

    const { getByTestId } = renderProvider(
      <ItunesWrapper dispatchSearchSongs={searchSongsSpy} dispatchClearGridData={clearGridDataSpy} />
    );

    fireEvent.change(getByTestId('search-bar'), {
      target: { value: 'a' }
    });
    await timeout(500);
    expect(searchSongsSpy).toBeCalled();
    fireEvent.change(getByTestId('search-bar'), {
      target: { value: '' }
    });
    await timeout(500);
    expect(clearGridDataSpy).toBeCalled();
  });

  it('should call dispatchSearchSongs on change', async () => {
    const { getByTestId } = renderProvider(<ItunesWrapper dispatchSearchSongs={mockDispatchSearch} />);
    fireEvent.change(getByTestId('search-bar'), {
      target: { value: 'Despacito' }
    });
    await timeout(500);
    expect(mockDispatchSearch).toBeCalled();
  });

  it('should update the loading state and render the Grid when gridData is available', () => {
    const gridData = {
      resultCount: 50,
      results: [{ songName: 'Song1' }, { songName: 'Song2' }]
    };
    const { getByTestId } = renderProvider(<ItunesWrapper gridData={gridData} />);
    expect(getByTestId('for')).toBeInTheDocument();
  });
});
