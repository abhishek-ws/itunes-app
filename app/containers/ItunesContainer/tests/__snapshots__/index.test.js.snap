// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItunesContainer Tests should render and match to the snapshot 1`] = `
<body>
  <div>
    <div
      class="ItunesContainer__Container-rg528l-1 cmmmEB"
    >
      <div
        class="ant-card ant-card-bordered ItunesContainer__CustomCard-rg528l-0 jha-dNF"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              My iTunes Store
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <p
            class="T__StyledText-gjlic1-0 demndm"
            data-testid="t"
          >
            Search your songs ...
          </p>
          <span
            class="ant-input-group-wrapper ant-input-search"
          >
            <span
              class="ant-input-wrapper ant-input-group"
            >
              <input
                class="ant-input"
                data-testid="search-bar"
                type="text"
                value=""
              />
              <span
                class="ant-input-group-addon"
              >
                <button
                  class="ant-btn ant-btn-icon-only ant-input-search-button"
                  type="button"
                >
                  <span
                    aria-label="search"
                    class="anticon anticon-search"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ItunesContainer__ResultContainer-rg528l-2 eeApxe"
    >
      <div
        class="ant-card ant-card-bordered ItunesContainer__CustomCard-rg528l-0 jzArEu"
        color="grey"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              List of songs will appear below
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <p
            class="T__StyledText-gjlic1-0 kYcyRq"
            data-testid="t"
          >
            Enter the song name you want to listen right now ...
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
